DictConstruct sets correct types

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#23171 DictConstruct sets correct types**

-

Differential Revision: [D16423640](https://our.internmc.facebook.com/intern/diff/D16423640/)

smessmer committed:
DictConstruct sets correct types

-

Differential Revision: [D16423640](https://our.internmc.facebook.com/intern/diff/D16423640/)

zdevito commented on torch/csrc/jit/register_prim_ops.cpp:
```diff
@@ -1018,8 +1018,21 @@ RegisterOperators reg(
              throw std::runtime_error(
                  "DictConstruct must have an even number of inputs");
            }
+           if (node->outputs().size() != 1) {
```
This should be assertions. At this point we can assume the instruction is well-formed.

smessmer committed:
Update on "DictConstruct sets correct types"


-

Differential Revision: [D16423640](https://our.internmc.facebook.com/intern/diff/D16423640/)
